### PR TITLE
Add SoftWare Heritage persistent IDentifier snapshots

### PR DESCRIPTION
**Pull request by Ericson2314**
See
https://docs.softwareheritage.org/devel/swh-model/persistent-identifiers.html
for the technical details of that spec.

Software Heritage has done a superb job promoting content addressing in
general, and their identifier scheme (SWHIDs, for short) in particular.
By supporting them in CIDs / IPLD, I hope the IPFS ecosystem can align
itself with that effort.

Per the linked documentation, SWHIDs have their own nested grammar and
versioning scheme. I begun by taking the version 1 core identifier
grammar, unrolled it, and replaced `:` with `-` per the guidelines on
separators, with the result being these 5 rows.

There is overlap between the remaining for and git-raw, so this just
adds SWHID snapshots for now.

### CODE REVIEW

**Review by vmx — APPROVED** [review]
Before merging I'd like to get an approval/thumbs up from other maintainers as well.

**Review by warpfork — APPROVED** [review]
I'm good with this.  :+1: 

I think no one has raised any objections to this code, in particular, at all -- so I'll merge this shortly if no objections appear.

**Pull request merged**
